chains/evm/executor/message-handler_test.go

```diff
@@ -5,20 +5,320 @@ package executor_test
 
 import (
 	"encoding/hex"
+	"errors"
 	"math/big"
 	"testing"
 
 	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/events"
-	"github.com/ChainSafe/chainbridge-core/chains/evm/executor/proposal"
-	"github.com/ChainSafe/chainbridge-core/relayer/message"
 
+	"github.com/ChainSafe/sygma-relayer/chains"
 	"github.com/ChainSafe/sygma-relayer/chains/evm/calls/contracts/bridge"
 	"github.com/ChainSafe/sygma-relayer/chains/evm/executor"
-	"github.com/ChainSafe/sygma-relayer/chains/evm/listener"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/suite"
 )
 
+var errIncorrectERC20PayloadLen = errors.New("malformed payload. Len  of payload should be 2")
+var errIncorrectERC721PayloadLen = errors.New("malformed payload. Len  of payload should be 3")
+var errIncorrectGenericPayloadLen = errors.New("malformed payload. Len  of payload should be 1")
+
+var errIncorrectAmount = errors.New("wrong payload amount format")
+var errIncorrectRecipient = errors.New("wrong payload recipient format")
+var errIncorrectTokenID = errors.New("wrong payload tokenID format")
+var errIncorrectMetadata = errors.New("wrong payload metadata format")
+
+// ERC20
+type Erc20HandlerTestSuite struct {
+	suite.Suite
+}
+
+func TestRunErc20HandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(Erc20HandlerTestSuite))
+}
+
+func (s *Erc20HandlerTestSuite) SetupSuite()    {}
+func (s *Erc20HandlerTestSuite) TearDownSuite() {}
+func (s *Erc20HandlerTestSuite) SetupTest()     {}
+func (s *Erc20HandlerTestSuite) TearDownTest()  {}
+
+func (s *Erc20HandlerTestSuite) TestErc20HandleMessage() {
+	message := &executor.TransferMessage{
+		Source:      1,
+		Destination: 0,
+		Data: executor.TransferMessageData{
+			DepositNonce: 1,
+			ResourceId:   [32]byte{0},
+			Payload: []interface{}{
+				[]byte{2}, // amount
+				[]byte{241, 229, 143, 177, 119, 4, 194, 218, 132, 121, 165, 51, 249, 250, 212, 173, 9, 147, 202, 107}, // recipientAddress
+			},
+		},
+	}
+
+	prop, err := executor.ERC20MessageHandler(message)
```

Comment:
Could you just call HandleMessage instead of directly the handlers?
They seem they could just be private anyway.